tasks/nunjucks.js

```diff
@@ -6,65 +6,159 @@
  * Licensed under the MIT license.
  */
 
-var nunjucks = require('nunjucks');
-var path = require('path');
-var async = require('async');
+'use strict'
 
-module.exports = function(grunt) {
-    'use strict';
+const nunjucks = require('nunjucks')
+const chalk    = require('chalk')
+const path     = require('path')
 
-    grunt.registerMultiTask('nunjucks', 'Renders nunjucks` template to HTML', function() {
-        var options = this.options();
-        var completeTask = this.async();
+module.exports = function (grunt) {
 
-        if (!options.data) {
-            grunt.log.warn('Template`s data is empty. Guess you forget to specify data option');
+    grunt.registerMultiTask('nunjucks', `Renders Nunjucks' templates to HTML`, function () {
+        // Declare async task
+        const completeTask = this.async()
+
+        // Get options and set defaults
+        // @note We're using `undefined` to fallback to Nunjucks' default settings
+        const options = this.options({
+            paths                : '',
+            autoescape           : undefined,
+            throwOnUndefined     : undefined,
+            trimBlocks           : undefined,
+            lstripBlocks         : undefined,
+            noCache              : undefined,
+            tags                 : undefined,
+            configureEnvironment : false,
+            data                 : false,
+            preprocessData       : false
+        })
+
+        // Finish task if no files specified
+        if (!this.files.length) {
+            grunt.log.error('No files specified.')
+
+            // Finish task — nothing we can do without specified files
+            return completeTask()
         }
 
-        var envOptions = { watch: false };
-        if (options.tags) {
-            envOptions.tags = options.tags;
+        // Warn in case of undefined data
+        if (!options.data) {
+            grunt.log.error(`Template's data is empty. Guess you've forget to specify data option.`)
         }
 
-        var basePath = options.paths || '';
-        var env = nunjucks.configure(basePath, envOptions);
+        // Arm Nunjucks
+        const env = nunjucks.configure(options.paths, {
+            watch            : false,
+            autoescape       : options.autoescape,
+            throwOnUndefined : options.throwOnUndefined,
+            trimBlocks       : options.trimBlocks,
+            lstripBlocks     : options.lstripBlocks,
+            noCache          : options.noCache,
+            tags             : options.tags
+        })
 
+        // Pass configuration to Nunjucks if specified
         if (typeof options.configureEnvironment === 'function') {
-            options.configureEnvironment.call(this, env, nunjucks);
+            options.configureEnvironment.call(this, env, nunjucks)
         }
 
-        async.each(this.files, function(f, done) {
-            var filepath = path.join(process.cwd(), f.src[0]);
+        // Get number of files
```

Comment:
I propose to leave them as they are and you will strip them in your own PR according to your vision